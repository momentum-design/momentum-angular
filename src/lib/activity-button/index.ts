export * from './activity-button.module';
export * from './activity-button.component';
