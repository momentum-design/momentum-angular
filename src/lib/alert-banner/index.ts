export * from './alert-banner.module';
export * from './alert-banner.component';
