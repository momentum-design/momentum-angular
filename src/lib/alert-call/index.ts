export * from './alert-call.module';
export * from './alert-call.component';
export * from './alert-call-config';
export * from './alert-call-container.component';
export * from './alert-call.service';
export * from './device-list-call.component';
