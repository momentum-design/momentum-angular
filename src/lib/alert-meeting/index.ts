export * from './alert-meeting.module';
export * from './alert-meeting.service';
export * from './alert-meeting-container.component';
export * from './alert-meeting-config';
export * from './alert-meeting.component';
