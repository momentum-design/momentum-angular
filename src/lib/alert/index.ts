export * from './alert.module';
export * from './alert.service';
export * from './alert-container.component';
export * from './alert-config';
export * from './alert.component';
