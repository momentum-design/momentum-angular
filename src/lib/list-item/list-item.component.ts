import {
  Component,
  OnInit,
  Input,
  HostBinding,
  Output,
  EventEmitter,
  ElementRef,
  HostListener,
  AfterViewInit
} from '@angular/core';
import {ENTER, SPACE, hasModifierKey} from '@angular/cdk/keycodes';
import { Highlightable } from '@angular/cdk/a11y';
import { uniqueId } from 'lodash';

export class OptionSelectionChange {
  constructor(
    public source: ListItemComponent,
    public isUserInput = false
  ) {}
}

@Component({
  selector: 'a[md-list-item], div[md-list-item], div[md-select-option]',
  exportAs: 'mdListItem',
  template: `
      <ng-container *ngIf="isSelectOption">

        <ng-container *ngIf="!isMulti">
          <md-list-item-section key="child-0" position='center'>
              <ng-container *ngIf="label; else content">{{label}}</ng-container>
              <ng-template #content>
                  <ng-content></ng-content>
              </ng-template>
          </md-list-item-section>

          <md-list-item-section key="child-1" position='right'>
              <md-icon *ngIf="selected" name="check_20" color="blue-50"></md-icon>
          </md-list-item-section>
        </ng-container>

        <ng-container *ngIf="isMulti">
          <md-checkbox
            name="{{id}}-checkbox"
            value="{{label}}"
            label="{{label}}"
            [(checkStatus)]="checkStatus"
            htmlId="{{id}}-checkbox">
          </md-checkbox>
        </ng-container>

      </ng-container>

      <ng-container *ngIf="!isSelectOption">
        <div *ngIf="label; else content">{{label}}</div>
        <ng-template #content>
          <ng-content></ng-content>
        </ng-template>
      </ng-container>
  `,
  host: {
    'class': 'md-list-item',
    '[class.md-list-item--read-only]': 'isReadOnly',
    '[class.md-list-item--separator]': 'separator',
    '[class.md-select-option]': 'isSelectOption',
    '[class.active]': 'active',
    '[class.focus]': 'focus',
    '[class.disabled]': 'disabled',
  }
})
export class ListItemComponent implements Highlightable, OnInit, AfterViewInit {
  checkedValues: string[] = [''];
  checkStatus: boolean;

  readonly isSelectOption: boolean = this._hasHostAttributes('md-select-option');

  constructor(public el: ElementRef) { }

  /** @prop Active prop to determine styles | false */
  @Input() active = false;
  /** @prop selected prop to determine styles | false */
  @Input() selected = false;
  /** @prop Disabled attribute for ListItem to determine styles | false */
  @Input() disabled = false;
  /** @prop Sets ListItem to a focus state | false */
  @Input() focus = false;
  /** @prop Sets ListItem id | null */
  @HostBinding('attr.id') @Input() id: string = uniqueId('md-list-item-');
  /** @prop Determines if is part of a multi select | false */
  @Input() isMulti = false;
  /** @prop Determines if ListItem is clickable | false */
  @Input() isReadOnly = false;
  /** @prop ListItem text and form value | '' */
  @Input() label = null;
  /** @prop external link associated input | '' */
  @Input() link = '';
  /** @prop ListItem ref name | 'navLink' */
  @Input() role = null;
  @HostBinding('attr.role') get myRole(): string {
    return this.role || this.isSelectOption ? 'option' : 'listItem';
  }
  /** @prop Controls whether to show separator or not | false */
  @Input() separator = false;
  /** @prop ListItem Title | '' */
  @Input() title = '';
  /** @prop ListItem size | '' */
  private _type: string = null;
  @Input()
  set type(type: string) {
    if (this._type) {
      this.el.nativeElement.classList.remove(`md-list-item--${this._type}`);
    }
    this.el.nativeElement.classList.add(`md-list-item--${type}`);
    this._type = type;
  }
  /** @prop ListItem value for OnSelect value | '' */
  @Input() value: string | number | object | any[] = '';

  @Output() selectionChange = new EventEmitter<OptionSelectionChange>();

  @HostBinding('attr.href') get myHref(): string {
    return (this.link && this._getHostElement().localName === 'a') ? this.link : null;
  }

  @HostBinding('attr.title') get theTitle() {
    return this.title || this.label;
  }

  @HostListener('click', ['$event']) handleClick = (event: MouseEvent) => {
    if (this.isReadOnly) {
      event.stopImmediatePropagation();
    } else {
      this._selectViaInteraction();
    }
  }

  @HostListener('keydown', ['$event']) handleKeydown = event => {
    if ((event.keyCode === ENTER || event.keyCode === SPACE) && !hasModifierKey(event)) {
      this._selectViaInteraction();
      event.preventDefault();
    }
  }

  ngOnInit() {
    if (this._type && !this._isTypeOptionValid()) {
      throw new Error(`md-list-item: ListItem type option must be one of the following:
        small, large, xlarge, space, header, 36, 52, 60`);
    }
  }

  ngAfterViewInit() {}

  select(): void {
    if (!this.selected) {
      this.selected = true;
      this._emitSelectionChangeEvent();
    }
  }

  /** Deselects the option. */
  deselect(): void {
    if (this.selected) {
      this.selected = false;
      this._emitSelectionChangeEvent();
    }
  }

  // Highlightable Interface methods
  setActiveStyles(): void {
    this.focus = true;
  }
  setInactiveStyles(): void {
    this.focus = false;
  }
  getLabel?(): string {
    return this.label;
  }

  _selectViaInteraction(): void {
    if (!this.disabled) {
      this.selected = !this.selected;
      this.checkStatus = this.selected;
      this._emitSelectionChangeEvent(true);
    }
  }

  _emitSelectionChangeEvent (isUserInput = false) {
    return this.selectionChange.emit(new OptionSelectionChange(this, isUserInput));
  }

  _isTypeOptionValid = () => (
    ['', 'small', 'large', 'xlarge', 'space', 'header', 36, 52, 60].includes(this._type)
  )

  _getHostElement() {
    return this.el.nativeElement;
  }

  /** Gets whether the button has one of the given attributes. */
  _hasHostAttributes(...attributes: string[]) {
    return attributes.some(attribute => this._getHostElement().hasAttribute(attribute));
  }
}
