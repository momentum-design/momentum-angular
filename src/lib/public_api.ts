/*
 * Public API Surface of collab-ui-angular
 */

export * from './alert';
export * from './alert-banner';
export * from './avatar';
export * from './badge';
export * from './composite-avatar';
export * from './button';
export * from './checkbox';
export * from './icon';
export * from './input-error';
export * from './input-helper';
export * from './input';
export * from './label';
export * from './list-item-section';
export * from './list-item';
export * from './list';
export * from './loading';
export * from './modal-body';
export * from './modal-footer';
export * from './modal-header';
export * from './modal';
export * from './radio';
export * from './spinner';
export * from './tabs';
export * from './toggle-switch';
