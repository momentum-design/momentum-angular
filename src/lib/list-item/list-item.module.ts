import { NgModule, CUSTOM_ELEMENTS_SCHEMA, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListItemComponent } from './list-item.component';
import { IconModule } from '../icon';
import { ListItemSectionModule} from '../list-item-section';
import { CheckboxModule } from '../checkbox';

@NgModule({
  imports: [
    CheckboxModule,
    CommonModule,
    IconModule,
    ListItemSectionModule
  ],
  declarations: [ListItemComponent],
  exports: [ListItemComponent],
  schemas: [CUSTOM_ELEMENTS_SCHEMA]
})
export class ListItemModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: ListItemModule,
      providers: [],
    };
  }
}
