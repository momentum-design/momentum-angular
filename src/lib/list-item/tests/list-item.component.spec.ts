import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ListItemComponent } from '../list-item.component';
import { CheckboxModule, IconModule, ListItemSectionModule } from 'src/lib/public_api';

@Component({
  template: `
    <div
      md-list-item
      [ngClass]="'my-ng-class'"
      class="custom-class"
      label='List Item A'
      [disabled]="true"
      [isReadOnly]="true"
      [separator]="true"
      title="my-title"
      type="small"
      ></div>
  `
})
class TestHostComponent {}

describe('ListItemComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ ListItemComponent, TestHostComponent ],
      imports: [CheckboxModule, IconModule, ListItemSectionModule]
    })
      .compileComponents();
  }));

  describe('main tests', () => {
    let component: ListItemComponent;
    let fixture: ComponentFixture<ListItemComponent>;
    let listItemNativeElement: HTMLElement;

    beforeEach(() => {
      fixture = TestBed.createComponent(ListItemComponent);
      component = fixture.componentInstance;
      component.label = 'myLabel';
      fixture.detectChanges();
    });

    it('should create and match snapshot', () => {
      expect(component).toBeTruthy();
      expect(fixture).toMatchSnapshot();
    });

    it('should render with disabled class when defined', () => {
      component.disabled = true;
      fixture.detectChanges();

      listItemNativeElement = fixture.nativeElement;
      expect(listItemNativeElement.className).toContain('disabled');
    });

    it('should render as readOnly when defined', () => {
      component.isReadOnly = true;
      fixture.detectChanges();

      listItemNativeElement = fixture.nativeElement;
      expect(listItemNativeElement.className).toContain(
        'md-list-item--read-only'
      );
    });
  });

  describe('with HostComponent', () => {
    let testHost: TestHostComponent;
    let fixture: ComponentFixture<TestHostComponent>;

    it('should pass classNames to the component', () => {
      fixture = TestBed.createComponent(TestHostComponent);
      testHost = fixture.componentInstance;
      const listItem = fixture.nativeElement.querySelector('.md-list-item');
      fixture.detectChanges();

      expect(fixture).toMatchSnapshot();
      expect(listItem.className).toContain(`md-list-item`);
      expect(listItem.className).toContain('my-ng-class');
      expect(listItem.className).toContain('custom-class');
    });

    it('should render the defined type', () => {
      fixture = TestBed.createComponent(TestHostComponent);
      testHost = fixture.componentInstance;
      const listItem = fixture.nativeElement.querySelector('.md-list-item');
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();

      expect(listItem.className).toContain(`md-list-item--small`);
    });

    it('should render the passed in props within testComponent template', () => {
      fixture = TestBed.createComponent(TestHostComponent);
      testHost = fixture.componentInstance;
      const listItem = fixture.nativeElement.querySelector('.md-list-item');
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();

      expect(listItem.className).toContain(`md-list-item--separator`);
      expect(listItem.className).toContain(`md-list-item--read-only`);
      expect(listItem.className).toContain(`disabled`);
    });
  });
});
