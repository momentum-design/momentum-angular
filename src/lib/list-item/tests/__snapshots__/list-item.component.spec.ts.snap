// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemComponent main tests should create and match snapshot 1`] = `
<a
  _isTypeOptionValid={[Function Function]}
  _type="null"
  active="false"
  checkedValues={[Function Array]}
  disabled="false"
  el={[Function ElementRef]}
  focus="false"
  handleClick={[Function Function]}
  handleKeydown={[Function Function]}
  id={[Function String]}
  isMulti="false"
  isReadOnly="false"
  isSelectOption="false"
  label={[Function String]}
  link=""
  role="null"
  selected="false"
  selectionChange={[Function EventEmitter]}
  separator="false"
  title=""
>
  <div>
    myLabel
  </div>
</a>
`;

exports[`ListItemComponent with HostComponent should pass classNames to the component 1`] = `
<ng-component>
  <div
    class="custom-class md-list-item my-ng-class md-list-item--small md-list-item--read-only md-list-item--separator disabled"
    md-list-item=""
    id="md-list-item-4"
    label="List Item A"
    ng-reflect-disabled="true"
    ng-reflect-is-read-only="true"
    ng-reflect-klass="custom-class"
    ng-reflect-label="List Item A"
    ng-reflect-ng-class="my-ng-class"
    ng-reflect-separator="true"
    ng-reflect-title="my-title"
    ng-reflect-type="small"
    role="listItem"
    title="my-title"
    type="small"
  >




    <div>
      List Item A
    </div>

  </div>
</ng-component>
`;

exports[`ListItemComponent with HostComponent should render the defined type 1`] = `
<ng-component>
  <div
    class="custom-class md-list-item my-ng-class md-list-item--small md-list-item--read-only md-list-item--separator disabled"
    md-list-item=""
    id="md-list-item-5"
    label="List Item A"
    ng-reflect-disabled="true"
    ng-reflect-is-read-only="true"
    ng-reflect-klass="custom-class"
    ng-reflect-label="List Item A"
    ng-reflect-ng-class="my-ng-class"
    ng-reflect-separator="true"
    ng-reflect-title="my-title"
    ng-reflect-type="small"
    role="listItem"
    title="my-title"
    type="small"
  >




    <div>
      List Item A
    </div>

  </div>
</ng-component>
`;

exports[`ListItemComponent with HostComponent should render the passed in props within testComponent template 1`] = `
<ng-component>
  <div
    class="custom-class md-list-item my-ng-class md-list-item--small md-list-item--read-only md-list-item--separator disabled"
    md-list-item=""
    id="md-list-item-6"
    label="List Item A"
    ng-reflect-disabled="true"
    ng-reflect-is-read-only="true"
    ng-reflect-klass="custom-class"
    ng-reflect-label="List Item A"
    ng-reflect-ng-class="my-ng-class"
    ng-reflect-separator="true"
    ng-reflect-title="my-title"
    ng-reflect-type="small"
    role="listItem"
    title="my-title"
    type="small"
  >




    <div>
      List Item A
    </div>

  </div>
</ng-component>
`;
